source/implementations/windows/Meadow.Windows/WindowsNetworkAdapterCollection.cs

```diff
@@ -135,7 +136,23 @@ private List<WifiNetwork> Scan()
 
     public AntennaType CurrentAntenna => throw new NotImplementedException();
 
-    public override bool IsConnected => throw new NotImplementedException();
+    public override bool IsConnected => IsInternetAvailable();
+
+    private bool IsInternetAvailable()
+    {
+        try
+        {
+            return NetworkInterface.GetIsNetworkAvailable() &&

```

Comment:
This also is looking at "all adapters" not just the wifi adapter.  The WiFi adapter could be off and this would return true if you had a wired network.

Reply:
Right, we want to validate if we have connectivity regardless of the adapter, right?